server/controllers/user-books-controller.js

```diff
@@ -0,0 +1,351 @@
+                                          Bookname: borrowedBook.bookName,
+                                          BookQuantity: borrowUpdate.bookQuantity,
+                                          BorrowedDate: lendUpdate.borrowDate,
+                                          DueDate: lendUpdate.dueDate,
+                                          returnDate: lendUpdate.actualReturnDate,
+                                          outStanding:
+                                          (lendUpdate.actualReturnDate - lendUpdate.dueDate) < 0 ?
+                                            0 : lendUpdate.actualReturnDate - lendUpdate.dueDate,
+                                        },
+                                      });
+                                    } else {
+                                      res.status(500).json({
+                                        status: 'Unsuccessful',
+                                        message: 'An error has occured',
+                                        l: lendUpdate,
+                                        b: borrowUpdate,
+                                      });
+                                    }
+                                  })
+                                  .catch(error => res.status(500).send(error));
+                              } else {
+                                res.status(404).json({
+                                  status: 'Unsuccessful',
+                                  message: 'Book not found',
+                                });
+                              }
+                            })
+                            .catch(error => res.status(500).send(error));
+                        } else {
+                          res.status(401).json({
+                            status: 'Unsuccessful',
+                            message: 'User/Book not matching records',
+                          });
+                        }
+                      } else if (foundLentBook.actualReturnDate !== null) {
+                        res.status(200).json({
+                          status: 'Unsuccessful',
+                          message: 'Book Already Returned',
+                        });
+                      } else {
+                        res.status(404).json({
+                          status: 'Unsuccessful',
+                          message: 'No records found',
+                        });
+                      }
+                    }).catch(error => res.status(500).send(error));
+                }
+              })
+              .catch(error => res.status(500).send(error));
+          }
+        } else {
+          res.status(501).json({
+            status: 'Unsuccessful',
+            message: 'Try again!??',
+          });
+        }
+      })
+      .catch(error =>
+        res.status(404).json({
+          status: 'Unsuccessful',
+          message: error,
+        }));
+  }
+}
+export default userBookInteraction;
```

Comment:
same here.Abstract re-useable code block to a function and call the function in your controller